File: DSA/1. Hungry Heroes/Part 1 (Easy)/Python/judge.py
from collections import Counter
from solution import Solution

class Judge:
    def run_tests(self):
        test_cases = [
            ["Beef", "Beef", "Beef"],
            ["Beef", "Beef", "Chicken", "Pork", "Vegetable", "Grain", "Vegetable", "Vegetable"],
            ["Fruit", "Grain", "Fruit", "Grain", "Beef"],
            []
        ]
        for test_case, expected in test_cases:
            self.test(test_case, expected)

    def test(self, stock, expected):
        solution = Solution()
        actual = solution.split_food(stock)
        passed = actual == expected
        print(f"Input: {stock}")
        print(f"Expected: {expected}")
        print(f"Actual: {actual}")
        print("Test " + ("PASSED" if passed else "FAILED"))
        print("-----------")

    def test_case_solution_output(self, stock):
        food_count = Counter(stock)
        result = []
        for food, count in food_count.items():
            result.append(food)
            result.append(str(count))
        return result

File: DSA/1. Hungry Heroes/Part 1 (Easy)/JavaScript/judge.js
const Solution = require('./solution');

class Judge {
    runTests() {
        const testCases = [
            ["Beef", "Beef", "Beef"],
            ["Beef", "Beef", "Chicken", "Pork", "Vegetable", "Grain", "Vegetable", "Vegetable"],
            ["Fruit", "Grain", "Fruit", "Grain", "Beef"],
            ["Fruit", "Beef", "Vegetable"],
            []
        ];

        testCases.forEach(testCase => {
            this.test(testCase, this.testCaseSolutionOutput(testCase));
        });
    }

    test(stock, expected) {
        const solution = new Solution();
        const actual = solution.splitFood(stock);
        const passed = JSON.stringify(actual) === JSON.stringify(expected);

        console.log(`Input: ${JSON.stringify(stock)}`);
        console.log(`Expected: ${JSON.stringify(expected)}`);
        console.log(`Actual: ${JSON.stringify(actual)}`);
        console.log(passed ? "Test PASSED" : "Test FAILED");
        console.log("-----------");
    }

    testCaseSolutionOutput(stock) {
        const foodCount = stock.reduce((acc, food) => {
            acc[food] = (acc[food] || 0) + 1;
            return acc;
        }, {});

        const result = [];
        for (const [food, count] of Object.entries(foodCount)) {
            result.push(food);
            result.push(count.toString());
        }
        return result;
    }
}

module.exports = Judge;